src/dds/proc.nix.cpp

```diff
@@ -75,8 +84,9 @@ proc_result dds::run_proc(const proc_options& opts) {
     auto rc            = ::pipe(stdio_pipe);
     check_rc(rc == 0, "Create stdio pipe for subprocess");
 
-    int read_pipe  = stdio_pipe[0];
-    int write_pipe = stdio_pipe[1];
+    int  read_pipe   = stdio_pipe[0];
+    auto read_closer = fd_closer{read_pipe};
+    int  write_pipe  = stdio_pipe[1];
```

Comment:
write_pipe is closed via a direct call to `::close` - I considered changing that to use the RAII closer too but went for minimal change. As it stands if `spawn_child` throws then this will leak an FD